perma_web/tasks/dev.py

```diff
@@ -1411,3 +1412,128 @@ def populate_benchmarked_wacz_sizes(ctx, source_csv):
     links = Link.objects.filter(guid__in=guids).values_list('guid', flat=True)
     for link in links.iterator():
         populate_wacz_size.delay(link)
+
+
+@task
+def sample_objects(ctx, n=1000):
+    """
+    Produce a sample of archive paths and etags for assessing completeness
+    of mirroring; because the comparison happens on the mirror, which doesn't
+    know anything about Perma, it produces a Python program to run on the
+    mirror.
+
+    The sample size, n, must be of a size to produce at least ten failures,
+    given an expected proportion of failures.
+    """
+    def get_etag(bucket, path):
+        return storages[bucket].connection.Object(
+            bucket_name=storages[bucket].bucket_name,
+            key=os.path.join(settings.MEDIA_ROOT, path)
+        ).e_tag.strip('"')
+
+    # obtain a sample of size n
+    links = Link.objects.filter(cached_can_play_back=True).order_by("?")[:n]
```

Comment:
I know we talked about `cached_can_play_back=True` offline, but in working on https://github.com/harvard-lil/perma/pull/3656, I realized that we DO in fact have a set of links where `cached_can_play_back` is wrong. Those links need their metadata fixed. They will cause problems here.

Reply:
Ah, thanks! I'll take a look at other options for filtering.

Reply:
I think we may have a few WACZ files where wacz_size is zero or not set?

Reply:
So the mirror might well have a WACZ file that the application doesn't consider a thing anymore. I think the WACZ in question would remain in the bucket as well? I'm not sure this makes a difference for this task; probably Perma's sense of what exists should be the ground truth, not what's in the bucket.

Reply:
Right, when things are replaced, the object remains in the bucket, I think on the principle that deleting things from the bucket is dangerous.